packages/lib/gpcircuits/circuits/tuple.circom

```diff
@@ -0,0 +1,64 @@
+pragma circom 2.1.8;
+
+include "circomlib/circuits/poseidon.circom"; // Poseidon hash
+include "gpc-util.circom"; // input selector
+
+/**
+ * Module mapping MAX_VALUES signals to MAX_TUPLES tuples of
+ * arity TUPLE_ARITY, each of which is encoded in the form of a
+ * Poseidon hash on MAX_ARITY elements. Lower-order tuples
+ * are assumed to have 0s in unoccupied slots, which may
+ * be accomplished by setting the corresponding tuple index
+ * to a sufficiently large number, e.g. -1 mod p.
+ */
+
+template TupleModule(
+    // Arity of each generated tuple
+    TUPLE_ARITY,
+    // Maximum number of value hashes to choose from
+    MAX_VALUES,
+    // Maxium number of tuples to generate
+    MAX_TUPLES
+) {
+    // Value hashes as input signals
+    signal input valueHash[MAX_VALUES];
```

Comment:
Same as in the list module, do these specifically need to be hashes, or are they just inputs where anything can work?